gobblin-config-management/gobblin-config-client/src/main/java/gobblin/config/client/ConfigClient.java

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.client;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.typesafe.config.Config;
+
+import gobblin.config.client.api.ConfigStoreFactoryDoesNotExistsException;
+import gobblin.config.client.api.VersionStabilityPolicy;
+import gobblin.config.common.impl.ConfigStoreBackedTopology;
+import gobblin.config.common.impl.ConfigStoreBackedValueInspector;
+import gobblin.config.common.impl.ConfigStoreTopologyInspector;
+import gobblin.config.common.impl.ConfigStoreValueInspector;
+import gobblin.config.common.impl.InMemoryTopology;
+import gobblin.config.common.impl.InMemoryValueInspector;
+import gobblin.config.store.api.ConfigKeyPath;
+import gobblin.config.store.api.ConfigStore;
+import gobblin.config.store.api.ConfigStoreCreationException;
+import gobblin.config.store.api.ConfigStoreFactory;
+import gobblin.config.store.api.ConfigStoreWithStableVersioning;
+import gobblin.config.store.api.VersionDoesNotExistException;
+
+/**
+ * This class is used by Client to access the Configuration Management core library.
+ * 
+ * 
+ * @author mitu
+ *
+ */
+public class ConfigClient {
+  private final VersionStabilityPolicy policy;
+
+  /** Normally key is the ConfigStore.getStoreURI(), value is the ConfigStoreAccessor
+   *  
+   * However, there may be two entries for a specific config store, for example
+   * if user pass in URI like "etl-hdfs:///datasets/a1/a2" and the etl-hdfs config store factory using
+   * default authority/default config store root normalized the URI to
+   * "etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest/datasets/a1/a2"
+   * 
+   * Then there will be two entries in the Map which point to the same value
+   * key1: "etl-hdfs:/"
+   * key2: "etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest/"
+   * 
+   */
+  private final TreeMap<URI, ConfigStoreAccessor> configStoreAccessorMap = new TreeMap<>();
+  
+  private final ConfigStoreFactoryRegister configStoreFactoryRegister;
+
+  private ConfigClient(VersionStabilityPolicy policy) {
+    this.policy = policy;
```

Comment:
Can you make this call `this(policy, new ConfiStoreFactoryRegister())` instead? It makes for easier maintanability in the future.